src/Lean/Server/Watchdog.lean

```diff
@@ -336,7 +338,7 @@ section ServerM
       | none    => (false, none)
     if pendingEdit then
       return
-    match fw.state with
+    match (fw.state) with
```

Comment:
Is this a workaround for some bug in Lean? We should not need parentheses here.

Reply:
no just a left over from a previous edit were I was using a `←` statement, I'm reverting the change.

Reply:
by the way you don't need parentheses even when you use `<-` in match. Speaking of which @leodemoura , I notice that it is fairly common to see `match (<- foo) with`  in the lean sources, is this a deliberate style choice or did the precedence change at some point? I usually leave the parentheses off in this situation.

Reply:
Yes, this is a deliberate style choice. I also like to avoid `(<-foo)` without a space between `<-` and `foo`. Sometimes I regret not enforcing the parentheses around `<- foo`. I would like the code in this repo to use a consistent style.

Reply:
Good to know, I'll keep that in mind.

Reply:
I notice we have some linting stuff happening, can this be added to the linter?

Reply:
I was just about to say the same thing. Could you open an issue to track this @lovettchris ? There are plenty of places to use a missing / confusing parentheses linter.

Reply:
Yes, it can be done with a linter, but please no linter PR unless it is almost perfect. This linter is very low priority.

Reply:
Just the issue for now, I think, so we don't forget. Style linters generally will need some design work.

Reply:
Yes, an issue is perfect.

Reply:
There's already this one, is this enough?
https://github.com/leanprover/lean4/issues/404